lib/internal/process.js

```diff
@@ -65,6 +65,22 @@ function setupConfig(_source) {
     if (value === 'false') return false;
     return value;
   });
+  // Intl.v8BreakIterator() without data crashes with a fatal error, so throw instead.
+  if (global.hasOwnProperty('Intl') &&
+      Intl.hasOwnProperty('v8BreakIterator') &&
+      process.hasOwnProperty('config') &&
+      process.config.hasOwnProperty('variables') &&
+      process.config.variables.icu_small &&
+      !process.icu_data_dir) {
```

Comment:
Bit of a challenge with this. It turns out that there is a significant amount of userland code that will completely overwrite `process.config` such that the internally set config properties won't be available.